src/services/blockchain/clients/midnight/client.rs

```diff
@@ -0,0 +1,246 @@
+			BlockFilterFactory,
+		},
+		filter::{evm_helpers::string_to_h256, MidnightBlockFilter},
+	},
+};
+
+/// Client implementation for Midnight blockchain
+///
+/// Provides high-level access to Midnight blockchain data and operations through HTTP transport.
+#[derive(Clone)]
+pub struct MidnightClient<T: Send + Sync + Clone> {
+	/// The underlying Midnight transport client for RPC communication
+	http_client: T,
+}
+
+impl<T: Send + Sync + Clone> MidnightClient<T> {
+	/// Creates a new Midnight client instance with a specific transport client
+	pub fn new_with_transport(http_client: T) -> Self {
+		Self { http_client }
+	}
+}
+
+impl MidnightClient<MidnightTransportClient> {
+	/// Creates a new Midnight client instance
+	///
+	/// # Arguments
+	/// * `network` - Network configuration containing RPC endpoints and chain details
+	///
+	/// # Returns
+	/// * `Result<Self, anyhow::Error>` - New client instance or connection error
+	pub async fn new(network: &Network) -> Result<Self, anyhow::Error> {
+		let http_client = MidnightTransportClient::new(network).await?;
+		Ok(Self::new_with_transport(http_client))
+	}
+}
+
+impl<T: Send + Sync + Clone + BlockchainTransport> BlockFilterFactory<Self> for MidnightClient<T> {
+	type Filter = MidnightBlockFilter<Self>;
+	fn filter() -> Self::Filter {
+		MidnightBlockFilter {
+			_client: PhantomData,
+		}
+	}
+}
+
+/// Extended functionality specific to Midnight blockchain
+#[async_trait]
+pub trait MidnightClientTrait {
+	/// Retrieves a transaction receipt by its hash
+	///
+	/// # Arguments
+	/// * `transaction_hash` - The hash of the transaction to look up
+	///
+	/// # Returns
+	/// * `Result<TransactionReceipt, anyhow::Error>` - Transaction receipt or error
+	async fn get_transaction_receipt(
+		&self,
+		transaction_hash: String,
+	) -> Result<EVMTransactionReceipt, anyhow::Error>;
+
+	/// Retrieves logs for a range of blocks
+	///
+	/// # Arguments
+	/// * `from_block` - Starting block number
+	/// * `to_block` - Ending block number
+	///
+	/// # Returns
+	/// * `Result<Vec<Log>, anyhow::Error>` - Collection of matching logs or error
+	async fn get_logs_for_blocks(
+		&self,
+		from_block: u64,
+		to_block: u64,
+	) -> Result<Vec<EVMReceiptLog>, anyhow::Error>;
+}
+
+#[async_trait]
+impl<T: Send + Sync + Clone + BlockchainTransport> MidnightClientTrait for MidnightClient<T> {
+	/// Retrieves a transaction receipt by hash with proper error handling
+	#[instrument(skip(self), fields(transaction_hash))]
+	async fn get_transaction_receipt(
+		&self,
+		transaction_hash: String,
+	) -> Result<EVMTransactionReceipt, anyhow::Error> {
+		let hash = string_to_h256(&transaction_hash)
+			.map_err(|e| anyhow::anyhow!("Invalid transaction hash: {}", e))?;
+
+		let params = json!([format!("0x{:x}", hash)])
+			.as_array()
+			.with_context(|| "Failed to create JSON-RPC params array")?
+			.to_vec();
+
+		let response = self
+			.http_client
+			.send_raw_request(
+				"eth_getTransactionReceipt",
```

Comment:
Is this a valid Midnight RPC method?

Reply:
Aha good catch, this was just a placeholder and is replaced by logic in this #218 PR